### PR TITLE
Unit testing data

### PR DESCRIPTION
**Pull request by EnderSmith**
TL;DR:
- ADDED `UnitTest()`s for all functions under // data manipulation
- ADDED global variable object for `diceRollUnitTests.js`
- REMOVED in code comments, in favor of `console.log()`
- REVERTED to `if (this.functionReturned !== true)` 
- - instead of `if (!this.functionReturned)` because the latter returned `true` even if the value returned was not of type bool 
- IMPROVED unit testing logging readability:
- - ADDED headers on different sections of unit tests
- - ADDED tabs where helpful in logging test results
- - ADDED `UnitTest.functionReturn` to display when test fails

also:
- REPLACED `==` with `===` in three places in `diceRoll2.js`

### CODE REVIEW

File: diceRoll/diceRollUnitTests.js [hunk 1]
```diff
@@ -46,14 +153,15 @@ function UnitTest(testName, functionToBeTested) {
   this.functionToBeTested = functionToBeTested;
   this.report;
   try {
-    if (!functionToBeTested()) {
```
Comment (EnderSmith) [inline diceRoll/diceRollUnitTests.js:49]: REVERTED, because this was approving functions that returned a string, treating it as a `true` statement

Comment (ultimatetag) [reply diceRoll/diceRollUnitTests.js:49]: I wouldn't overload the response of the unit test to multiple responsibilities (1. signaling pass/fail and 2. reporting the reason).

This is why other frameworks pass some kind of object into their tests to operate on. simplest version from here would be something like:
```
var mytest = new UnitTest('passed test', function(context) {
   context.report = 'reason';
   return false;
}
```
more complicated would be something like having an assert object that has an assert.result property that it sets based on functions you execute on it and you check that after the test runs. (meaning the tests don't actually return anything at all.)
```
new UnitTest('test', function(assert) {
   assert.ok(true|false, 'reason');
   assert.fail('reason');
   assert.equal(1, 1, 'reason');
   assert.notEqual(1, 2, 'reason');
});
```


example (not tested):

```
function Assert(testname) {
   this.result = true;
   this.log = 'test: ' + testname;
   function ok(result, reason) {
      if (result) {
         this.log += '\npassed: ok: ' + reason;
      } else {
         this.result = false;
         this.log = '\nfailed: ok:' + reason;
      }
   }
}
function UnitTest(testname, test) {
  this.testname = testname;
  this.test = test;
  function run() {
    var assert = new Assert(testname);
    try {
       this.test(assert);
    } catch (ex) {
       assert.result = false;
       assert.reason = '\nexception: ' + ex;
    }
    return assert;
  }
}
var mytest = new UnitTest('testname', function(assert) {
  assert.ok(false, 'failed'); 
});

var testrun = mytest.run();
if (!testrun.result) {
   console.log(mytest.testname + ' failed. log: ' + testrun.log);
} else {
   console.log(mytest.testname + ' passed. log: ' + testrun.log);
}

```

**Review by ultimatetag — APPROVED** [review]
approve with suggestions

File: diceRoll/diceRollUnitTests.js (hunk 1 above)
Comment (EnderSmith) [reply diceRoll/diceRollUnitTests.js:49]: I don't understand enough of what's going on with that code to fix what's wrong with it. Pulling this file as is for now. Maybe you can explain this to me later.

**Comment by EnderSmith** [comment]
TL;DR:
- ADDED `UnitTest()`s for all functions under // data manipulation
- ADDED global variable object for `diceRollUnitTests.js`
- REMOVED in code comments, in favor of `console.log()`
- REVERTED to `if (this.functionReturned !== true)` 
- - instead of `if (!this.functionReturned)` because the latter returned `true` even if the value returned was not of type bool 
- IMPROVED unit testing logging readability:
- - ADDED headers on different sections of unit tests
- - ADDED tabs where helpful in logging test results
- - ADDED `UnitTest.functionReturn` to display when test fails

also:
- REPLACED `==` with `===` in three places in `diceRoll2.js`
